app/lib/actions/middleware/record_smart_proxy_sync_history.rb

```diff
@@ -0,0 +1,17 @@
+module Actions
+  module Middleware
+    class RecordSmartProxySyncHistory < Dynflow::Middleware
+      def run(*args)
+        smart_proxy_history_ids = []
+        if (action.input[:repository_id] && (action.input[:smart_proxy_id] || action.input[:capsule_id]) && !self.action.output[:smart_proxy_history_ids])
+          repo = ::Katello::Repository.find(action.input[:repository_id])
+          smart_proxy_id = action.input[:smart_proxy_id] || action.input[:capsule_id]
+          smart_proxy = ::SmartProxy.find(smart_proxy_id)
+          smart_proxy_history_ids << repo.create_smart_proxy_sync_history(smart_proxy)
+          self.action.output[:smart_proxy_history_ids] = smart_proxy_history_ids
```

Comment:
a little niptick, i think only one id can exist, so maybe rename  output[:smart_proxy_history_ids] to smart_proxy_history_id ?